Brokerages/Alpaca/AlpacaBrokerage.Utility.cs

```diff
@@ -0,0 +1,428 @@
+                startTime = DateTimeHelper.FromUnixTimeMilliseconds(offset);
+            }
+        }
+
+        /// <summary>
+        /// Aggregates a list of trade ticks into tradebars
+        /// </summary>
+        /// <param name="symbol">The symbol</param>
+        /// <param name="ticks">The IEnumerable of ticks</param>
+        /// <param name="period">The time span for the resolution</param>
+        /// <returns></returns>
+        internal static IEnumerable<TradeBar> AggregateTicks(Symbol symbol, IEnumerable<Tick> ticks, TimeSpan period)
+        {
+            return
+                from t in ticks
+                group t by t.Time.RoundDown(period)
+                into g
+                select new TradeBar
+                {
+                    Symbol = symbol,
+                    Time = g.Key,
+                    Open = g.First().LastPrice,
+                    High = g.Max(t => t.LastPrice),
+                    Low = g.Min(t => t.LastPrice),
+                    Close = g.Last().LastPrice,
+                    Volume = g.Sum(t => t.Quantity),
+                    Period = period
+                };
+        }
+
+        /// <summary>
+        /// Converts an Alpaca order into a LEAN order.
+        /// </summary>
+        private Order ConvertOrder(IOrder order)
+        {
+            var type = order.OrderType;
+
+            Order qcOrder;
+            switch (type)
+            {
+                case Markets.OrderType.Stop:
+                    qcOrder = new StopMarketOrder
+                    {
+                        StopPrice = order.StopPrice.Value
+                    };
+                    break;
+
+                case Markets.OrderType.Limit:
+                    qcOrder = new LimitOrder
+                    {
+                        LimitPrice = order.LimitPrice.Value
+                    };
+                    break;
+
+                case Markets.OrderType.StopLimit:
+                    qcOrder = new StopLimitOrder
+                    {
+                        Price = order.StopPrice.Value,
+                        LimitPrice = order.LimitPrice.Value
+                    };
+                    break;
+
+                case Markets.OrderType.Market:
+                    qcOrder = new MarketOrder();
+                    break;
+
+                default:
+                    throw new NotSupportedException(
+                        "An existing " + type + " working order was found and is currently unsupported. Please manually cancel the order before restarting the algorithm.");
+            }
+
+            var instrument = order.Symbol;
+            var id = order.OrderId.ToString();
+
+            qcOrder.Symbol = Symbol.Create(instrument, SecurityType.Equity, Market.USA);
+
+            if (order.SubmittedAt != null)
+            {
+                qcOrder.Time = order.SubmittedAt.Value;
+            }
+
+            qcOrder.Quantity = order.Quantity;
```

Comment:
Should the `Alpaca.Order.Side` be used to transform the quantity to negative, as done when doing the opposite conversion at `GenerateAndPlaceOrder`?